pkg/metamorph/client.go

```diff
@@ -294,28 +340,70 @@ func (m *Metamorph) SubmitTransactions(ctx context.Context, txs sdkTx.Transactio
 
 		return ret, nil
 	}
+	var responses *metamorph_api.TransactionStatuses
+	var err error
 
 	// put all transactions together
-	start := time.Now()
-	const interval = 300 * time.Millisecond
-	maxTimeout := time.Duration(5 * time.Second)
-	if len(in.Transactions) != 0 {
-		maxTimeout = time.Duration(in.Transactions[0].MaxTimeout) * time.Second
+	maxTimeout := max(time.Duration(in.Transactions[0].MaxTimeout)*time.Second, m.maxTimeout)
+
+	retryTicker := time.NewTicker(retryInterval)
+
+	timeoutTimer := time.NewTimer(maxTimeout)
+
```

Comment:
Move this after the first  `m.client.PutTransactions(ctx, request)` call. Add `defer retryTicker.Stop()` to prevent leaks. (I'm not sure if `timeoutTimer` also needs to be stopped manually.)